app/src/main/java/com/android/shelfLife/model/user/UserRepositoryFirestore.kt

```diff
@@ -234,93 +251,17 @@ class UserRepositoryFirestore(
     updateUserField("email", email)
   }
 
   override suspend fun updateSelectedHousehold(selectedHouseholdUID: String) {
     updateUserField("selectedHouseholdUID", selectedHouseholdUID)
   }
 
-  override fun getUserIds(userEmails: Set<String?>, callback: (Map<String, String>) -> Unit) {
-    if (userEmails.isEmpty()) {
-      callback(emptyMap())
-      return
-    }
-    val emailBatches = userEmails.chunked(10) // Firestore allows up to 10 values in 'whereIn'
-    val emailToUserId = mutableMapOf<String, String>()
-    var batchesProcessed = 0
-
-    for (emailBatch in emailBatches) {
-      db.collection("users")
-          .whereIn("email", emailBatch)
-          .get()
-          .addOnSuccessListener { querySnapshot ->
-            for (doc in querySnapshot.documents) {
-              val email = doc.getString("email")
-              val userId = doc.id
-              if (email != null) {
-                emailToUserId[email] = userId
-              }
-            }
-            batchesProcessed++
-            if (batchesProcessed == emailBatches.size) {
-              callback(emailToUserId)
-            }
-          }
-          .addOnFailureListener { exception ->
-            Log.e("HouseholdRepository", "Error fetching user IDs by emails", exception)
-            batchesProcessed++
-            if (batchesProcessed == emailBatches.size) {
-              callback(emailToUserId)
-            }
-          }
-    }
-  }
-
-  override fun getUserEmails(userIds: List<String>, callback: (Map<String, String>) -> Unit) {
-    if (userIds.isEmpty()) {
-      callback(emptyMap())
-      return
-    }
-
-    val uidBatches = userIds.chunked(10) // Firestore allows up to 10 values in 'whereIn'
-    val uidToEmail = mutableMapOf<String, String>()
-    var batchesProcessed = 0
-
-    for (uidBatch in uidBatches) {
-      db.collection("users")
-          .whereIn(FieldPath.documentId(), uidBatch)
-          .get()
-          .addOnSuccessListener { querySnapshot ->
-            for (doc in querySnapshot.documents) {
-              val email = doc.getString("email")
-              val userId = doc.id
-              if (email != null) {
-                uidToEmail[userId] = email
-              }
-            }
-            batchesProcessed++
-            if (batchesProcessed == uidBatches.size) {
-              callback(uidToEmail)
-            }
-          }
-          .addOnFailureListener { exception ->
-            Log.e("HouseholdRepository", "Error fetching emails by user IDs", exception)
-            batchesProcessed++
-            if (batchesProcessed == uidBatches.size) {
-              callback(uidToEmail)
-            }
-          }
-    }
-  }
-
-  override suspend fun selectHousehold(household: HouseHold?) {
-    _selectedHousehold.value = household
-    household?.let { updateSelectedHousehold(it.uid) }
+  override suspend fun selectHousehold(householdUid: String?) {
+    householdUid?.let { updateSelectedHousehold(it) }
```

Comment:
since the selectedHousehold is no longer a variable of the repository, but one from the houshold repository, you correctly use the function from the household repo